File: backend/PaulFunBlogger.Api/Entities/Category.cs
namespace PaulFunBlogger.Api.Entities;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public int? ParentId { get; set; }
    public int SortOrder { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public Category? Parent { get; set; }
    public ICollection<Category> Children { get; set; } = new List<Category>();
    public ICollection<Article> Articles { get; set; } = new List<Article>();
}

File: backend-go/internal/models/category.go
package models

type Category struct {
	ID        uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Name      string    `gorm:"not null;size:100" json:"name"`
	Slug      string    `gorm:"uniqueIndex;not null;size:100" json:"slug"`
	ParentID  *uint     `gorm:"index" json:"parentId"`
	SortOrder int       `gorm:"default:0" json:"sortOrder"`
	Articles  []Article `gorm:"foreignKey:CategoryID" json:"-"`
}
